src/agent/communicator/src/https_verifier_win.cpp

```diff
@@ -0,0 +1,198 @@
+#ifndef WIN32_LEAN_AND_MEAN
+#define WIN32_LEAN_AND_MEAN
+#endif
+
+// clang-format off
+#include <windows.h>
+#include <wincrypt.h>
+// clang-format on
+
+#include "https_verifier_win.hpp"
+#include "cert_store_utils_win.hpp"
+#include "icert_store_utils_win.hpp"
+#include "ix509_utils.hpp"
+#include "x509_utils.hpp"
+#include <logger.hpp>
+
+#include <boost/asio/ssl.hpp>
+
+#include <memory>
+#include <string>
+
+namespace https_socket_verify_utils
+{
+    bool HttpsVerifierWin::Verify(boost::asio::ssl::verify_context& ctx)
+    {
+        if (!m_x509Utils || !m_certStoreUtils)
```

Comment:
Isn't this validation in macOS necessary?

Reply:
Added!